src/ralph/backends/data/ldp.py

```diff
@@ -0,0 +1,244 @@
+
+        Args:
+            target (str or None): The target stream_id where to list the archives.
+                If target is `None`, the `DEFAULT_STREAM_ID` is used instead.
+            details (bool): Get detailed archive information instead of just ID's.
+            new (bool): Given the history, list only not already read archives.
+
+        Yields:
+            str: If `details` is False.
+            dict: If `details` is True.
+
+        Raises:
+            BackendParameterException: If the `target` is `None` and no
+                `DEFAULT_STREAM_ID` is given.
+            BackendException: If a failure during retrieval of archives list occurs.
+        """
+        list_archives_endpoint = self.get_archive_endpoint(stream_id=target)
+        logger.debug("List archives endpoint: %s", list_archives_endpoint)
+        logger.debug("List archives details: %s", str(details))
+
+        try:
+            archives = self.client.get(list_archives_endpoint)
+        except ovh.exceptions.APIError as error:
+            msg = "Failed to get archives list: %s"
+            logger.error(msg, error)
+            raise BackendException(msg % error) from error
+
+        logger.debug("Found %d archives", len(archives))
+
+        if new:
+            archives = set(archives) - set(self.get_command_history(self.name, "read"))
+            logger.debug("New archives: %d", len(archives))
+
+        if not details:
+            for archive in archives:
+                yield archive
+
+            return
+
+        for archive in archives:
+            yield self._details(target, archive)
+
+    @enforce_query_checks
+    def read(
+        self,
+        *,
+        query: Union[str, BaseQuery] = None,
+        target: str = None,
+        chunk_size: Union[None, int] = 4096,
+        raw_output: bool = True,
+        ignore_errors: bool = True,
+    ) -> Iterator[Union[bytes, dict]]:
+        """Reads an archive matching the query in the target stream_id and yields it.
+
+        Args:
+            query: (str or BaseQuery): The ID of the archive to read.
+            target (str or None): The target stream_id containing the archives.
+                If target is `None`, the `DEFAULT_STREAM_ID` is used instead.
+            chunk_size (int or None): The chunk size for reading archives.
+            raw_output (bool): Ignored. Always set to `True`.
+            ignore_errors (bool): Ignored. Always set to `True`.
+
+        Yields:
+            bytes: The content of the archive matching the query.
+
+        Raises:
+            BackendException: If a failure during the read operation occurs.
+            BackendParameterException: If the `query` argument is not an archive name.
+        """
+        if query.query_string is None:
+            msg = "Invalid query. The query should be a valid archive name"
+            raise BackendParameterException(msg)
+
+        if not raw_output or not ignore_errors:
```

Comment:
For now, only reading raw (binary) output is implemented.
Do we decompress the (.gz/.zip/.7z or .zst and potentially PGP encrypted) archive and JSON parse the (GELF or plain) content if `raw_output` is `False`?